### PR TITLE
Set FlagNetStandard1XDependencies=true and fix issues

### PR DESCRIPTION
**Pull request by akoeplinger**
The only product change is upgrading Microsoft.DiaSymReader from 1.4.0 to 2.0.0 in ILCompiler.TypeSystem.csproj, the rest are test changes.

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
     <!-- FluentAssertions library has a generic recursion and the warning breaks NativeAOT build -->
     <NoWarn>$(NoWarn);IL3054</NoWarn>
+
+    <!-- Microsoft.DotNet.ProjectModel brings in NS1.x dependencies -->
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: Curious, what is Microsoft.DotNet.ProjectModel? Is that a library that we don't produce live anymore?

Comment (akoeplinger) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: yeah it is some old library from project.json days I think: https://www.nuget.org/packages/Microsoft.DotNet.ProjectModel/

I'm not sure the tests that use it provide much value so we might consider deleting the code.

Comment (akoeplinger) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: This source file specifically: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextBuilderTests.cs

Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: @ericstj had a quick chat about this. The `DependencyContextBuilder` type that is used in those tests lives in that `Microsoft.DotNet.ProjectModel` package but at some point then got moved into dotnet/sdk: https://github.com/dotnet/sdk/blob/bedb8ccb29ebda43103d0f1559106db704e15b52/src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs#L14

SDK doesn't reference the ProjectModel package at all and it's already marked as deprecated on nuget.org: https://www.nuget.org/packages/Microsoft.DotNet.ProjectModel/1.0.0-rc3-1-003177#versions-body-tab

I'm skeptical that we still need the tests but if we do, we should move them into dotnet/sdk next to the existing ones: https://github.com/dotnet/sdk/blob/bedb8ccb29ebda43103d0f1559106db704e15b52/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs. That said, I don't think that's necessary. They should definitely be deleted from here though as the ProjectModel package 

cc @eerhardt who moved DependencyModel into runtime: https://github.com/dotnet/runtime/commit/b93829cf38f079ccb895708bb7e938ab9541bcca

Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: See comment in https://github.com/dotnet/runtime/pull/34296#discussion_r401855280. Copied here:

> Both Microsoft.DotNet.InternalAbstractions and Microsoft.DotNet.ProjectModel are obsolete packages (think project.json and .xproj time frame). The issue is our unit tests are written relying on these old packages, and we never took the time to rewrite the tests. But we don't want to lose the test coverage.

Comment (ericstj) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:13]: I'm not so sure on that - this single file `DependencyContextBuilderTests.cs` is the only one using those old references.  AFAICT every test here is just calling `DependencyContextBuilder.Build` and making sure the results are as expected.  That's mostly testing `ProjectModel` and not `DependencyModel`.  Sure `DependencyContextBuilder` returns types defined in `DependencyModel` but I'm not sure if we're getting much out of these tests that we don't get out of the unit tests for those types and APICompat.

File: src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
      <!-- some tests require full ICU data, force it -->
     <WasmIncludeFullIcuData>true</WasmIncludeFullIcuData>
+
+    <!-- Microsoft.CodeAnalysis.CSharp.CodeFix.Testing brings in NS1.x dependencies -->
```
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:16]: cc @jaredpar

[the remark this answers is not in the document]
Comment (jaredpar) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:16]: That comes from [roslyn-sdk][1] which is a gray area repository. Not really part of the product but used in the product. 

[1]: https://github.com/dotnet/roslyn-sdk

[the remark this answers is not in the document]
Comment (akoeplinger) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj]: There are three dependencies which cause this, all coming from Microsoft.CodeAnalysis.Analyzer.Testing.

Upgrading these three resolves the NS1.x dependencies so I committed that:
```xml
<PackageReference Include="DiffPlex" Version="1.7.2" />
<PackageReference Include="Microsoft.VisualBasic" Version="10.2.0" />
<PackageReference Include="Microsoft.VisualStudio.Composition" Version="17.0.46" />
```

File: eng/Versions.props [hunk 3]
```diff
@@ -186,14 +189,15 @@
     <MicrosoftDotNetXHarnessCLIVersion>9.0.0-prerelease.24066.3</MicrosoftDotNetXHarnessCLIVersion>
     <MicrosoftDotNetHotReloadUtilsGeneratorBuildToolVersion>9.0.0-alpha.0.24065.1</MicrosoftDotNetHotReloadUtilsGeneratorBuildToolVersion>
     <NUnitVersion>3.12.0</NUnitVersion>
-    <NUnitTestAdapterVersion>4.1.0</NUnitTestAdapterVersion>
+    <NUnit3TestAdapterVersion>4.5.0</NUnit3TestAdapterVersion>
```
Comment (ViktorHofer) [inline eng/Versions.props:192]: Where is that property used?

File: src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj]: I see couple of issues in that dependency chain:

- Microsoft.CodeAnalysis.Analyzer.Testing

image

Microsoft.VisualBasic and System.ValueTuple shouldn't be referenced. They are already provided inbox. Does it really need all the NuGet dependencies?

- Microsoft.CodeAnalysis.Workspaces.Common

image

Microsoft.Bcl.AsyncInterfaces shouldn't be referenced. We already fixed that in main.

- Microsoft.CodeAnalysis.Common

image

System.Runtime.CompilerServices.Unsafe shouldn't be referenced for net7.0. It's part of the shared framework starting with .NET 7.

**Review by ViktorHofer — APPROVED** [review]
This is great 💯

**Pull request merged**